solana-programs/anchor/audius-data/tests/audius-data.ts

```diff
@@ -483,38 +470,114 @@ describe('audius-data', () => {
       bumpSeed,
       metadata,
       newUserAcctPDA
-    )
+    );
 
     // New sol key that will be used to permission user updates
-    let newUserKey = anchor.web3.Keypair.generate()
+    let newUserKey = anchor.web3.Keypair.generate();
 
     // Generate signed SECP instruction
     // Message as the incoming public key
-    let message = newUserKey.publicKey.toString()
+    let message = newUserKey.publicKey.toString();
 
     await testInitUserSolPubkey({
       message,
       pkString,
       privKey,
       newUserKey,
-      newUserAcctPDA
-    })
+      newUserAcctPDA,
+    });
 
-    let newTrackKeypair = anchor.web3.Keypair.generate()
+    let newTrackKeypair = anchor.web3.Keypair.generate();
 
-    let trackMetadata = randomCID()
+    let trackMetadata = randomCID();
     await testCreateTrack({
       trackMetadata,
       newTrackKeypair,
       userAuthorityKey: newUserKey,
-      trackOwnerPDA: newUserAcctPDA
-    })
+      trackOwnerPDA: newUserAcctPDA,
+      adminStgKeypair,
+    });
 
     await testDeleteTrack({
       trackKeypair: newTrackKeypair,
       trackOwnerPDA: newUserAcctPDA,
-      userAuthorityKey: newUserKey
-    })
+      userAuthorityKey: newUserKey,
+    });
+  });
+
+  it("create multiple tracks in parallel", async () => {
+    let {
+      privKey,
+      pkString,
+      testEthAddr,
+      testEthAddrBytes,
+      handleBytesArray,
+      metadata,
+    } = initTestConstants();
 
-  })
+    let { baseAuthorityAccount, bumpSeed, derivedAddress } =
+      await findDerivedPair(
+        program.programId,
+        adminStgKeypair.publicKey,
+        Buffer.from(handleBytesArray)
+      );
+    let newUserAcctPDA = derivedAddress;
+
+    await testInitUser(
+      baseAuthorityAccount,
+      testEthAddr,
+      testEthAddrBytes,
+      handleBytesArray,
+      bumpSeed,
+      metadata,
+      newUserAcctPDA
+    );
+
+    // New sol key that will be used to permission user updates
+    let newUserKey = anchor.web3.Keypair.generate();
+
+    // Generate signed SECP instruction
+    // Message as the incoming public key
+    let message = newUserKey.publicKey.toString();
+
+    await testInitUserSolPubkey({
+      message,
+      pkString,
+      privKey,
+      newUserKey,
+      newUserAcctPDA,
+    });
+
+    let newTrackKeypair = anchor.web3.Keypair.generate();
+    let newTrackKeypair2 = anchor.web3.Keypair.generate();
+    let newTrackKeypair3 = anchor.web3.Keypair.generate();
+    let trackMetadata = randomCID();
+    let trackMetadata2 = randomCID();
+    let trackMetadata3 = randomCID();
+    let start = Date.now();
+    await Promise.all([
+      testCreateTrack({
```

Comment:
Kinda a nit, but I would prefer that the creations be done and assertions be done after. It's hard when parsing the test to know where the creation happens vs. the test assertion that the creation did happen